File: CS/PersistentObjects.cs
namespace AssociatedCollectionFilter {
    using System;
    using DevExpress.Xpo;
    
    
	[Persistent("Customers")]
	public class Customer : XPLiteObject {
        public Customer(Session session) : base(session) { }
		[Key]
		public string CustomerID;
		public string CompanyName;
		public string ContactTitle;

		[Association("CustomerOrders", typeof(Order))]
		public XPCollection Orders {
			get {
				return GetCollection("Orders");
			}
		}
	}

	[Persistent("Orders")]
	public class Order : XPLiteObject {
        public Order(Session session) : base(session) { }

		[Key]
		public int OrderID;
		
		[Persistent("CustomerID"), Association("CustomerOrders")]
		public Customer Customer;

		[Persistent("EmployeeID"), Association("EmployeeOrders")]
		public Employee Employee;

        public DateTime OrderDate;
        public decimal Freight;
	}

	[Persistent("Employees")]
	public class Employee : XPLiteObject {
        public Employee(Session session) : base(session) { }
		[Key]
		public int EmployeeID;
		public string FirstName;
		public string LastName;

		[Association("EmployeeOrders", typeof(Order))]
		public XPCollection Orders {
			get {
				return GetCollection("Orders");
			}
		}
	}
}

File: VB/PersistentObjects.vb
Imports Microsoft.VisualBasic
	Imports System
	Imports DevExpress.Xpo
Namespace AssociatedCollectionFilter


	<Persistent("Customers")> _
	Public Class Customer
		Inherits XPLiteObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		<Key> _
		Public CustomerID As String
		Public CompanyName As String
		Public ContactTitle As String

		<Association("CustomerOrders", GetType(Order))> _
		Public ReadOnly Property Orders() As XPCollection
			Get
				Return GetCollection("Orders")
			End Get
		End Property
	End Class

	<Persistent("Orders")> _
	Public Class Order
		Inherits XPLiteObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		<Key> _
		Public OrderID As Integer

		<Persistent("CustomerID"), Association("CustomerOrders")> _
		Public Customer As Customer

		<Persistent("EmployeeID"), Association("EmployeeOrders")> _
		Public Employee As Employee

		Public OrderDate As DateTime
		Public Freight As Decimal
	End Class

	<Persistent("Employees")> _
	Public Class Employee
		Inherits XPLiteObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		<Key> _
		Public EmployeeID As Integer
		Public FirstName As String
		Public LastName As String

		<Association("EmployeeOrders", GetType(Order))> _
		Public ReadOnly Property Orders() As XPCollection
			Get
				Return GetCollection("Orders")
			End Get
		End Property
	End Class
End Namespace
